apps/dashboard/app/(app)/settings/billing/plans/button.tsx

```diff
@@ -47,6 +48,18 @@ export const ChangePlanButton: React.FC<Props> = ({ workspace, newPlan, label })
     },
   });
 
+  const handleClick = () => {
+    const hasPaymentMethod = !!workspace.stripeCustomerId;
```

Comment:
Only checking if there's a customer associated because I can't configure the stripe env within the client side component, and I want to avoid passing stripe info as a prop from the server page. 

this is where the edge case can occur: if they have a stripe customer id, but they've deleted their payment method from stripe, it won't error out until the trpc procedure attempts to change the plan, so it's just a UX inconvenience. the likelihood that the user will delete their payment method is very slim.